### PR TITLE
07-MemoryManagementUserSpace: add user space program

### PR DESCRIPTION
**Pull request by prosolvo4th**
Signed-off-by: yuriy.podgorniy <yuriy.pidgorniy@gmail.com>

### CODE REVIEW

File: 07-MemoryManagementUserSpace/main.c [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	{
+		ret.tv_nsec += 1000000000;
+		ret.tv_sec -= 1;
+	}
+	return ret;
+}
+
+const int testsNum = 3;
+const char *label[] = {"malloc", "calloc", "alloca"};
+void *ptr;
```
Comment (yekovalyov) [inline 07-MemoryManagementUserSpace/main.c]: Is it required to make ptr static?

**Comment by yekovalyov** [comment]
Please add results of program execution.

**Comment by prosolvo4th** [comment]
Changes: add results of runnig

File: 07-MemoryManagementUserSpace/main.c [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		for(int oversize = 0; oversize <= 1; oversize++)
+		{
+			for (int method = 0; method < testsNum; method++)
+			{
+				struct timespec t1, t2;
+				if(method == 0)
+				{
+					t1 = get_timestamp();
+					ptr = malloc(size + oversize);
+					t2 = get_timestamp();
```
Comment (yekovalyov) [inline 07-MemoryManagementUserSpace/main.c]: Is it possible to move these line from inner loop body.

**Comment by yekovalyov** [comment]
Please address all comments and provide execution result for each function in the different files.

File: 07-MemoryManagementUserSpace/main.c (hunk 1 above)
Comment (prosolvo4th) [reply 07-MemoryManagementUserSpace/main.c]: Yes, it must be static.

**Comment by prosolvo4th** [comment]
Changes: rewrite main.c, take out implementation of test into stand-alone function, add results

File: 07-MemoryManagementUserSpace/examples [hunk 3]
```diff
@@ -0,0 +1,137 @@
+the4th@the4th-Aspire-A715-72G:~/Projects/GL/myFork/gl-kernel-training-2019.git/07-MemoryManagementUserSpace$ ./main
```
Comment (yekovalyov) [inline 07-MemoryManagementUserSpace/examples]: This file is outdated. Please remove it from repository.

File: 07-MemoryManagementUserSpace/main.c (hunk 1 above)
Comment (yekovalyov) [reply 07-MemoryManagementUserSpace/main.c]: Is it static in the last version of your code?

**Comment by prosolvo4th** [comment]
Changes: removed outdated file

**Comment by prosolvo4th** [comment]
Changes: removed outdated file

**Pull request merged**